src/main/java/org/jabref/gui/JabRefFrame.java

```diff
@@ -262,7 +266,44 @@ private void initDragAndDrop() {
                             for (Tab libraryTab : tabbedPane.getTabs()) {
                                 if (libraryTab.getId().equals(tab.getId()) &&
                                         !tabbedPane.getSelectionModel().getSelectedItem().equals(libraryTab)) {
-                                    ((LibraryTab) libraryTab).dropEntry(stateManager.getLocalDragboard().getBibEntries());
+                                    LibraryTab libraryTabCasted = (LibraryTab) libraryTab;
+                                    if (dragboard.hasContent(DragAndDropDataFormats.GROUP)) {
+                                        List<String> groupPathToSources = (List<String>) dragboard.getContent(DragAndDropDataFormats.GROUP);
+                                        if (libraryTabCasted.getBibDatabaseContext().getMetaData().getGroups().isEmpty()) {
+                                            // a root (all entries) GroupTreeNode
+                                            GroupTreeNode rootNode = getCurrentLibraryTab().getBibDatabaseContext()
+                                                                                           .getMetaData()
+                                                                                           .getGroups()
+                                                                                           .get()
+                                                                                           .copyNode();
+
+                                            // add rootNode to the Library if it does not have a root.
+                                            libraryTabCasted.getBibDatabaseContext()
+                                                                     .getMetaData()
+                                                                     .setGroups(rootNode);
+                                        }
+                                        Optional<GroupTreeNode> parentGroup = Optional.of(libraryTabCasted
```

Comment:
The same comment as the above. It could be renamed to *destinationLibraryGroupRoot*

Reply:
Will be edited, thanks.